### PR TITLE
Make <delete> repeatable in Normal Mode. Fix #394

### LINKED ISSUE #394: delete N characters under and after the cursor with Del key
### Issue Summary

Now we only support `x` to delete characters while we can use `Del` on Windows/Linux platform to archive the target.

We only need to register this key in contribution and map it to `x` operator. I'd like to mark it as `easy` and `help_wanted`!

### PR DESCRIPTION
**Pull request by octref**
Yay! We love PRs! 🎊

Please include a description of your change & check your PR against this list, thanks:

- [x] Commit message has a short title & issue references
- [x] Commits are squashed 
- [x] It builds and tests pass (e.g `gulp tslint`)

Previously `<delete>` key works in Normal Mode but is not repeatable.
This PR makes it repeatable, so commands like `5<delete>` would work like `5x`.

### CODE REVIEW

**Comment by octref** [comment]
BTW, there are a few unrelated tests failing, but they also fail on the master branch without my changes. So I ignored them.

File: src/actions/actions.ts [hunk 1]
```diff
@@ -1960,6 +1960,11 @@ class ActionDeleteChar extends BaseCommand {
 }
 
 @RegisterAction
+class ActionDeleteCharWithDeleteKey extends ActionDeleteChar {
+  keys = ["<delete>"];
+}
```
Comment (rebornix) [inline src/actions/actions.ts]: There difference between delete and x is the way they handle numeric prefix. `Nx` will perform `x N times but `N<Del>` does nothing.

File: package.json [hunk 2]
```diff
@@ -53,6 +53,11 @@
                 "when": "editorTextFocus"
             },
             {
+                "key": "Delete",
+                "command": "extension.vim_delete",
+                "when": "editorTextFocus"
```
Comment (rebornix) [inline package.json]: We are reimplementing delete on in Normal mode, so you may want to check Vim state here like https://github.com/VSCodeVim/Vim/blob/master/package.json#L68

File: src/actions/actions.ts (hunk 1 above)
Comment (octref) [inline src/actions/actions.ts]: I thought `N<Del>` to delete the next 5 chars should be the expected behavior.

So, in Normal Mode, is this

`<Del>` -> removes next char
`N<Del>` -> do nothing

what I should do instead?

File: package.json (hunk 2 above)
Comment (octref) [inline package.json]: Thanks for the hint!

File: src/actions/actions.ts (hunk 1 above)
Comment (rebornix) [inline src/actions/actions.ts]: Yup, that's Vim's behavior and it's awkward, but let's do this.

File: test/mode/normalModeTests/commands.test.ts [hunk 3]
```diff
@@ -61,13 +68,20 @@ suite("Mode Normal", () => {
...
 
     newTest({
-      title: "Can handle x at end of line",
+      title: "Can handle 'x' at end of line",
       start: ['one tw|o'],
       keysPressed: '^llxxxxxxxxx',
       end: ['|'],
     });
 
     newTest({
```
Comment (rebornix) [inline test/mode/normalModeTests/commands.test.ts]: Also add a case for `N<delete>` as the behavior is strange in Vim.

File: src/actions/actions.ts [hunk 4]
```diff
@@ -1960,6 +1960,19 @@ class ActionDeleteChar extends BaseCommand {
 }
 
 @RegisterAction
+class ActionDeleteCharWithDeleteKey extends ActionDeleteChar {
+  keys = ["<delete>"];
+
+  public async execCount(position: Position, vimState: VimState): Promise<VimState> {
+    if (vimState.recordedState.count !== 0) {
```
Comment (johnfn) [inline src/actions/actions.ts]: This is actually impossible, as 0 by itself is a motion, so it can't prefix any motion.

**Comment by johnfn** [comment]
Fix up my one suggestion and we should be good to go here! Thanks for the PR, @octref!

File: src/actions/actions.ts (hunk 4 above)
Comment (octref) [inline src/actions/actions.ts]: It seems to me when I press `<delete>`, it goes to `execCount` with a 0 count (default value for an int).
In your BaseCommand, you have this line to set it to 1 if it's 0:
`let timesToRepeat = this.canBePrefixedWithCount ? vimState.recordedState.count || 1 : 1;`

Comment (johnfn) [inline src/actions/actions.ts]: The fact that Ndelete does nothing is so bizarre that I totally misread this. This is fine.

**Comment by johnfn** [comment]
Ah, I'm sorry for doing this retroactively, but you caught us in the middle of a refactor. 

Would you mind fixing this up by making both delete classes extend from an abstract base class rather than using inheritance? I'm trying to move away from big inheritance trees; see #519.

**Comment by octref** [comment]
No worries. Refactoring is good.
In work now. Will do that tonight.

**Comment by octref** [comment]
@johnfn 
Done. Sorry for the delay. Had some company events for the last few days.

**Comment by johnfn** [comment]
LGTM. Thanks, @octref!

**Pull request merged**